File: python/longest_palindromic_substring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        if not s or len(s) <= 1:
            return s
        res = ''
        for i in range(0, len(s)):
            palindrome = self.getPalindromicStr(s, i, i)
            if len(palindrome) > len(res):
                res = palindrome

            palindrome = self.getPalindromicStr(s, i, i + 1)
            if len(palindrome) > len(res):
                res = palindrome
        return res

    def getPalindromicStr(self, s, start, end):
        while(start >= 0 and end < len(s) and s[start] == s[end]):
            start -= 1
            end += 1
        return s[start + 1 : end]

File: java/Longest Palindromic Substring.java
public class Solution {
    public String longestPalindrome(String s) {
        if (s == null || s.length() == 0)
            return null;

        int length = s.length();
        String res = "";
        for (int i = 0; i < length; i++) {
            /*回文字符串为奇数*/
            String palindrome = getPalindrome(s, i, i);
            if (palindrome.length() > res.length()) {
                res = palindrome;
            }
            
            /*回文字符串为偶数*/
            palindrome = getPalindrome(s, i, i + 1);
            if (palindrome.length() > res.length()) {
                res = palindrome;
            }
        }

        return res;
    }

    public String getPalindrome(String s, int left, int right) {
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            left--;
            right++;
        }
        return s.substring(left + 1, right);
    }
}
